crates/matrix-sdk/src/encryption/recovery/mod.rs

```diff
@@ -288,8 +300,10 @@ impl Recovery {
         // Why oh why, can't we delete account data events?
         self.client.account().set_account_data(SecretStorageDisabledContent {}).await?;
         self.client.account().set_account_data(BackupDisabledContent { disabled: true }).await?;
+        // Delete all the known secrets we have in the account data.
+        self.delete_all_known_secrets().await?;
```

Comment:
Just before that we deleted `m.secret_storage.default_key`. Here we delete the secrets.
We should probably also delete the key account data itself? the one with type `m.secret_storage.key.ID-OF-THE-DEFAULT-KEY`

Reply:
Is there a point of deleting this? We can't remove the event as there isn't a `DELETE` API for this, so we replace sensitive things with an empty JSON content `{}`.

But the `m.secret_storage.key.ID-OF-THE-DEFAULT-KEY` events, do not contain sensitive data. They only contain information about the key which can be used, once you have the key, to check that it indeed is correct.

Reply:
> But the `m.secret_storage.key.ID-OF-THE-DEFAULT-KEY` events, do not contain sensitive data. They only contain information about the key which can be used, once you have the key, to check that it indeed is correct.

this is fair. I guess it might be nice just to clean it up a bit as best we can?

Reply:
ok, but you've done it anyway?

Reply:
Yes, sorry. I attempted to make a joke.